pkg/chain/upkeep_key.go

```diff
@@ -0,0 +1,31 @@
+package chain
+
+import (
+	"encoding/json"
+	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+	"strings"
+)
+
+type UpkeepKey []byte
+
+func (u UpkeepKey) BlockKeyAndUpkeepID() (types.BlockKey, types.UpkeepIdentifier) {
```

Comment:
Judging by the findings of the recent audit, this should probably return an error as well.